src/generator_files.rs

```diff
@@ -65,13 +51,23 @@ fn enumerate_files(
         .collect();
 
     // If shuffle is set, shuffle the files
-    let files_iter = if source_config.random_sampling {
+    if source_config.random_sampling {
         let mut rng = rand::rng(); // Get a random number generator, thread local. We don´t seed, so typically won't be reproducible
-        files_list.shuffle(&mut rng);
-        files_list.into_iter()
-    } else {
-        files_list.into_iter()
-    };
+        files_list.shuffle(&mut rng); // This happens in place
+    }
+
+    // If world_size > 1, we need to split the files list into chunks and only process the chunk corresponding to the rank
+    if source_config.world_size > 1 {
+        let chunk_size =
+            (files_list.len() as f64 / source_config.world_size as f64).floor() as usize;
+        let start = source_config.rank * chunk_size;
+        let mut end = start + chunk_size;
+        if source_config.rank == (source_config.world_size - 1) {
```

Comment:
ceil() removes this branch (and 3 lines), but adds a min(), taste and colors. Works for me in any case, not super important